docs/analyzer/developing_recognizers.md

```diff
@@ -1,3 +1,73 @@
+For tools and documentation on evaluating and analyzing recognizers, refer to the [presidio-research Github repository](https://www.github.com/presidio-research).
+
+> When contributing recognizers to the Presidio OSS, new predefined recognizers should be added to the [supported entities list](../supported_entities.md), and follow the [contribution guidelines](../../CONTRIBUTING.MD).
+
+### Performance
+
+Make sure your recognizer doesn't take too long to process text. Anything above 100ms per request with 100 tokens is probably not good enough.
+
+### Environment
+
+When adding new recognizers that have 3rd party dependencies, make sure that the new dependencies don't interfere with Presidio's dependencies. In the case of a conflict, one can create an isolated model environment (outside the main presidio-analyzer process) and implement a [`RemoteRecognizer`](../../presidio-analyzer/presidio_analyzer/remote_recognizer.py) on the presidio-analyzer side to interact with the model's endpoint. In addition, make sure the license on the 3rd party dependency allows you to use it for any purpose.
+
+## Recognizer Types
+
+Generally speaking, there are three types of recognizers:
+
+### Deny Lists
+
+A deny list is a list of words that should be removed during text analysis. For example, it can include a list of titles (`["Mr.", "Mrs.", "Ms.", "Dr."]` to detect a "Title" entity.)
+
+See [this documentation](index.md#how-to-add-a-new-recognizer) on adding a new recognizer. The [`PatternRecognizer`](/presidio-analyzer/presidio_analyzer/pattern_recognizer.py) class has built-in support for a deny-list input.
+
+### Pattern Based
+
+Pattern based recognizers use regular expressions to identify entities in text.
+See [this documentation](/docs/custom_fields.md#via-code) on adding a new recognizer via code.
+The [`PatternRecognizer`](/presidio-analyzer/presidio_analyzer/pattern_recognizer.py) class should be extended.
+See some examples here:
+
+> Examples of pattern based recognizers are the [`CreditCardRecognizer`](../../presidio-analyzer/presidio_analyzer/predefined_recognizers/credit_card_recognizer.py) and [`EmailRecognizer`](../../presidio-analyzer/presidio_analyzer/predefined_recognizers/email_recognizer.py).
+
+### Machine Learning (ML) Based or Rule-Based
+
+Many PII entities are undetectable using naive approaches like deny-lists or regular expressions.
+In these cases, we would wish to utilize a Machine Learning model capable of identifying entities in free text, or a rule-based recognizer. There are four options for adding ML and rule based recognizers:
+
+#### Utilize SpaCy or Stanza
+
+Presidio currently uses [spaCy](https://spacy.io/) as a framework for text analysis and Named Entity Recognition (NER), and [stanza](https://stanfordnlp.github.io/stanza/) as an alternative. As to not introduce new tools for no reason, it is recommended to first try to use `spaCy` or `stanza` over other tools.
```

Comment:
```suggestion
Presidio currently uses [spaCy](https://spacy.io/) as a framework for text analysis and Named Entity Recognition (NER), and [stanza](https://stanfordnlp.github.io/stanza/) as an alternative. To avoid introducing new tools, it is recommended to first try to use `spaCy` or `stanza` over other tools if possible.
```